2020-10-24 12:49:44.6622 INFO Books - GetBooks: Attempted GET.
2020-10-24 12:49:46.2865 INFO Books - GetBooks: Successful GET.
2020-10-24 12:50:03.3406 INFO Books - GetBook: Attempted GET with Id:1.
2020-10-24 12:50:03.5800 WARN Books - GetBook: GET Id:1 not found.
2020-10-24 13:06:20.7114 INFO Authors - Update: Attempted UPDATE with Id:4.
2020-10-24 13:06:20.7670 WARN Authors - Update: No data to UPDATE Id:4.
2020-10-24 13:07:09.5178 INFO Authors - Update: Attempted UPDATE with Id:4.
2020-10-24 13:07:10.9593 WARN Authors - Update: Not found Id:4.
2020-10-24 13:27:52.8692 INFO Books - Create: Attempted.
2020-10-24 13:27:53.0814 ERROR Missing type map configuration or unsupported mapping.

Mapping types:
Object -> Book
System.Object -> BookStore_API.Data.Book - 
2020-10-24 13:28:10.8809 INFO Books - Create: Attempted.
2020-10-24 13:28:10.9794 ERROR Missing type map configuration or unsupported mapping.

Mapping types:
Object -> Book
System.Object -> BookStore_API.Data.Book - 
2020-10-24 13:28:26.2968 INFO Books - Create: Attempted.
2020-10-24 13:28:26.3903 ERROR Missing type map configuration or unsupported mapping.

Mapping types:
Object -> Book
System.Object -> BookStore_API.Data.Book - 
2020-10-24 13:30:34.1442 INFO Books - Create: Attempted.
2020-10-24 13:30:36.9030 ERROR An error occurred while updating the entries. See the inner exception for details. - Microsoft.Data.SqlClient.SqlException (0x80131904): The INSERT statement conflicted with the FOREIGN KEY constraint "FK_Books_Authors". The conflict occurred in database "BookStore", table "dbo.Authors", column 'Id'.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__164_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:76f119ab-e562-43fb-9dd9-c652d451bb14
Error Number:547,State:0,Class:16
2020-10-24 13:31:12.1637 INFO Books - Create: Attempted.
2020-10-24 13:31:13.4183 ERROR An error occurred while updating the entries. See the inner exception for details. - Microsoft.Data.SqlClient.SqlException (0x80131904): The INSERT statement conflicted with the FOREIGN KEY constraint "FK_Books_Authors". The conflict occurred in database "BookStore", table "dbo.Authors", column 'Id'.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__164_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:76f119ab-e562-43fb-9dd9-c652d451bb14
Error Number:547,State:0,Class:16
2020-10-24 13:31:28.3969 INFO Books - Create: Attempted.
2020-10-24 13:31:29.5877 ERROR An error occurred while updating the entries. See the inner exception for details. - Microsoft.Data.SqlClient.SqlException (0x80131904): The INSERT statement conflicted with the FOREIGN KEY constraint "FK_Books_Authors". The conflict occurred in database "BookStore", table "dbo.Authors", column 'Id'.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__164_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:76f119ab-e562-43fb-9dd9-c652d451bb14
Error Number:547,State:0,Class:16
2020-10-24 13:39:33.1159 INFO Books - Create: Attempted.
2020-10-24 13:39:34.6197 INFO Books - Create: Successful.
2020-10-24 13:51:21.9752 INFO Books - Create: Attempted.
2020-10-24 13:51:23.5662 INFO Books - Create: Successful.
2020-10-24 13:51:23.5838 INFO Books - Create: BookStore_API.Data.Book
2020-10-24 13:53:42.1475 INFO Authors - GetAuthors: Attempted.
2020-10-24 13:53:42.5131 INFO Authors - GetAuthors: Successful.
2020-10-24 14:47:58.9553 INFO Books - GetBooks: Attempted.
2020-10-24 14:47:59.5050 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 14:49:13.9484 INFO Books - GetBook: Attempted with Id:1.
2020-10-24 14:49:14.0916 WARN Books - GetBook: Not found Id:1.
2020-10-24 14:49:37.2099 INFO Books - GetBook: Attempted with Id:2.
2020-10-24 14:49:37.2530 WARN Books - GetBook: Not found Id:2.
2020-10-24 14:50:09.3196 INFO Books - GetBook: Attempted with Id:4.
2020-10-24 14:50:09.7481 ERROR Books - GetBook: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 14:51:10.1886 INFO Authors - GetAuthors: Attempted.
2020-10-24 14:51:10.2338 INFO Authors - GetAuthors: Successful.
2020-10-24 14:51:20.4580 INFO Books - GetBooks: Attempted.
2020-10-24 14:51:20.8693 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 14:51:40.7160 INFO Books - GetBooks: Attempted.
2020-10-24 14:51:41.0966 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 14:53:20.8291 INFO Books - Create: Attempted.
2020-10-24 14:53:20.9021 INFO Books - Create: Successful.
2020-10-24 14:53:20.9254 INFO Books - Create: BookStore_API.Data.Book
2020-10-24 14:54:34.8461 INFO Books - GetBooks: Attempted.
2020-10-24 14:54:35.2501 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 14:59:41.5515 INFO Books - GetBooks: Attempted.
2020-10-24 14:59:43.4167 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 15:18:49.1746 INFO Books - GetBooks: Attempted.
2020-10-24 15:18:50.9860 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 15:21:16.6513 INFO Books - GetBooks: Attempted.
2020-10-24 15:21:18.5349 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 15:25:20.7739 INFO Books - GetBooks: Attempted.
2020-10-24 15:25:22.6347 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 15:31:08.0179 INFO Books - GetBooks: Attempted.
2020-10-24 15:31:09.8294 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 15:32:47.9885 INFO Books - GetBooks: Attempted.
2020-10-24 15:32:48.4289 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 16:56:41.3615 INFO Books - GetBooks: Attempted.
2020-10-24 16:56:42.7483 INFO Books - GetBooks: Successful.
2020-10-24 16:58:25.1472 INFO Books - GetBooks: Attempted.
2020-10-24 16:58:25.9721 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 16:59:06.5766 INFO Books - GetBook: Attempted with Id:1.
2020-10-24 16:59:06.9024 WARN Books - GetBook: Not found Id:1.
2020-10-24 16:59:15.4259 INFO Books - GetBook: Attempted with Id:7.
2020-10-24 16:59:16.0245 ERROR Books - GetBook: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 17:00:24.8920 INFO Books - Create: Attempted.
2020-10-24 17:00:25.3849 INFO Books - Create: Successful.
2020-10-24 17:00:25.4113 INFO Books - Create: BookStore_API.Data.Book
2020-10-24 17:01:31.5357 INFO Books - Update: Attempted with Id:8.
2020-10-24 17:01:33.6599 ERROR An error occurred while updating the entries. See the inner exception for details. - Microsoft.Data.SqlClient.SqlException (0x80131904): The UPDATE statement conflicted with the FOREIGN KEY constraint "FK_Books_Authors". The conflict occurred in database "BookStore", table "dbo.Authors", column 'Id'.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__164_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:5d3438aa-2fd2-4aff-89b0-354d346fe2a7
Error Number:547,State:0,Class:16
2020-10-24 17:05:36.1408 INFO Books - GetBooks: Attempted.
2020-10-24 17:05:39.0957 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 17:05:56.5801 INFO Books - Update: Attempted with Id:8.
2020-10-24 17:05:59.0580 ERROR An error occurred while updating the entries. See the inner exception for details. - Microsoft.Data.SqlClient.SqlException (0x80131904): The UPDATE statement conflicted with the FOREIGN KEY constraint "FK_Books_Authors". The conflict occurred in database "BookStore", table "dbo.Authors", column 'Id'.
The statement has been terminated.
   at Microsoft.Data.SqlClient.SqlCommand.<>c.<ExecuteDbDataReaderAsync>b__164_0(Task`1 result)
   at System.Threading.Tasks.ContinuationResultTaskFromResultTask`2.InnerInvoke()
   at System.Threading.Tasks.Task.<>c.<.cctor>b__274_0(Object obj)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
--- End of stack trace from previous location where exception was thrown ---
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot, Thread threadPoolThread)
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReaderAsync(RelationalCommandParameterObject parameterObject, CancellationToken cancellationToken)
   at Microsoft.EntityFrameworkCore.Update.ReaderModificationCommandBatch.ExecuteAsync(IRelationalConnection connection, CancellationToken cancellationToken)
ClientConnectionId:3735e04b-1671-405d-bf8d-56743e28fb2d
Error Number:547,State:0,Class:16
2020-10-24 17:18:15.2881 INFO Books - GetBooks: Attempted.
2020-10-24 17:18:17.1560 ERROR Books - GetBooks: Unable to cast object of type 'System.Decimal' to type 'System.Double'. - 
2020-10-24 17:19:37.5504 INFO Books - GetBooks: Attempted.
2020-10-24 17:19:39.1468 INFO Books - GetBooks: Successful.
2020-10-24 17:20:46.7594 INFO Books - Update: Attempted with Id:8.
2020-10-24 17:20:47.1284 INFO Books - Update: Successful with Id:8.
2020-10-24 17:21:17.3014 INFO Books - GetBooks: Attempted.
2020-10-24 17:21:17.3680 INFO Books - GetBooks: Successful.
2020-10-24 17:32:08.0190 INFO Books - GetBooks: Attempted.
2020-10-24 17:32:09.4959 INFO Books - GetBooks: Successful.
2020-10-24 17:32:34.1899 INFO Books - Delete: Attempted with Id:8.
2020-10-24 17:32:34.5788 INFO Books - Delete: Successful with Id:8.
2020-10-24 17:32:42.3145 INFO Books - GetBooks: Attempted.
2020-10-24 17:32:42.3508 INFO Books - GetBooks: Successful.
